types/utils.go

```diff
@@ -29,3 +29,25 @@ func CheckForDuplicatesAndEmptyStrings(input []string) error {
 
 	return nil
 }
+
+// Validator is an interface for types that have a Validate method.
+type Validator interface {
+	Validate() error
+}
+
+// ValidateEntries checks for duplicates based on a key extracted by keyFunc and validates each entry.
+func ValidateEntries[T Validator, K comparable](entries []T, keyFunc func(T) K) error {
```

Comment:
very nice function, I just don't like the file name being `utils` :sweat_smile:

Reply:
maybe `validate.go` ?

Reply:
yes, it's very generic. I can move this to a `validate.go` inside the `types` dir. Wdyt? do you have other suggestion?

Reply:
Sounds good to move into `types/validate.go`